File: CPP/Program/Barang.cpp
#include <iostream>
// include file kategori
#include "Kategori.cpp"

using namespace std;

class Barang : public Kategori { // inheritance dari class Kategori
    // atribut private
    private:
        string merk;
        string model;
        int tahun_produksi;

    public:
        // constructor kosong
        Barang(){}

        // constructor
        Barang(string id_kategori, string nama_kategori, string deskripsi, 
            string merk, string model, int tahun_produksi) : Kategori(id_kategori, nama_kategori, deskripsi) { // manggil constructor parent
            this->merk = merk;
            this->model = model;
            this->tahun_produksi = tahun_produksi;
        }

        // getter
        string getMerk() const {
            return merk;
        }
        string getModel() const {
            return model;
        }
        int getTahunProduksi() const {
            return tahun_produksi;
        }

        // setter
        void setMerk(string merk) {
            this->merk = merk;
        }
        void setModel(string model) {
            this->model = model;
        }
        void setTahunProduksi(int tahun_produksi) {
            this->tahun_produksi = tahun_produksi;
        }

        // destructor
        ~Barang() { }
};

File: Python/Program/Barang.py
# include Kategori.py
from Kategori import Kategori

class Barang(Kategori):
    def __init__(self, id_kategori="", nama_kategori="", deskripsi="",
                 merk="", model="", tahun_produksi=0):
        super().__init__(id_kategori, nama_kategori, deskripsi)
        self.__merk = merk
        self.__model = model
        self.__tahun_produksi = tahun_produksi

    # getter
    def get_merk(self):
        return self.__merk
    def get_model(self):
        return self.__model
    def get_tahun_produksi(self):
        return self.__tahun_produksi

    # setter
    def set_merk(self, merk):
        self.__merk = merk
    def set_model(self, model):
        self.__model = model
    def set_tahun_produksi(self, tahun_produksi):
        self.__tahun_produksi = tahun_produksi
